code/modules/mod/modules/modules_maint.dm

```diff
@@ -33,12 +33,12 @@
 	playsound(src, 'sound/items/modsuit/springlock.ogg', 75, TRUE)
 	addtimer(CALLBACK(src, .proc/snap_shut), rand(3 SECONDS, 5 SECONDS))
 	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, .proc/on_activate_spring_block)
+	UnregisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
```

Comment:
what if someone is turning off their suit while this happens? wouldnt it runtime from 2 unregister signals of the same type?

Reply:
Thats the problem

Reply:
i need to solve that